rust_src/src/math.rs

```diff
@@ -291,3 +291,101 @@ fn abs(arg: LispObject) -> LispObject {
         }
     }
 }
+
+#[repr(C)]
+#[derive(Clone, Copy)]
+pub enum ArithComparison {
+    ArithEqual,
+    ArithNotequal,
+    ArithLess,
+    ArithGrtr,
+    ArithLessOrEqual,
+    ArithGrtrOrEqual,
+}
+
+#[no_mangle]
+pub extern "C" fn arithcompare(
+    num1: LispObject,
+    num2: LispObject,
+    comparison: ArithComparison,
+) -> LispObject {
+
+    let obj1 = check_number_coerce_marker(num1);
+    let obj2 = check_number_coerce_marker(num2);
+
+    let result = if obj1.is_float() || obj2.is_float() {
```

Comment:
Also, this is sadly not exactly the logic that `arithcompare` uses on master. You can look at my version of `minmax_driver` in #227 (where `arithcompare` is basically inlined) to see how it should be done now.